django/utils/functional.py

```diff
@@ -147,19 +118,43 @@ def __eq__(self, other):
                 other = other.__cast()
             return self.__cast() == other
 
+        def __ne__(self, other):
```

Comment:
[Python 3 defaults `__ne__` as the opposite of `__eq__`](https://docs.python.org/3/reference/datamodel.html#object.__ne__). So I think we could drop this method.

Reply:
If the wrapped type implements `__ne__`, it might differ from `not __eq__`.

The previous code had made this assumption, including by using `@total_ordering`. But I think it's better not to.

Reply:
Excellent point! Thanks for explaining that. I agree that keeping this method is the right thing to do.